### PR TITLE
Catch submit errors

### PR DESCRIPTION
**Pull request by julianburr**
Rn we're not reseting the submit blocker properly if the submit function itself throws. This fixes it 🤞

### CODE REVIEW

File: src/react-forms.js [hunk 1]
```diff
@@ -416,20 +416,24 @@ class ReactForms extends Component {
...
+        await this.setErrors(errors, false, true);
+      } else {
+        await this.setErrors(maybePromisedErrors, false, true);
+      }
 
-    const result = await this.executeSubmit();
-    this.blockSubmission = false;
-    return result;
+      const result = await this.executeSubmit();
+      this.blockSubmission = false;
```
Comment (ababkov) [inline src/react-forms.js]: Could shove this in a finally block 🤷‍♀

Comment (aaronnuu) [reply src/react-forms.js]: Actually good point is it better to shove this in a catch-less finally block and let the error fall through? Are we just ignoring submit errors now? @julianburr

Comment (julianburr) [reply src/react-forms.js]: you can't bail on catch/finally if you use async/await, or am I misunderstanding what you guys mean?

Comment (julianburr) [reply src/react-forms.js]: or do you mean use `.finally` where we execute `this.submitForm`? In general finally is not (yet) part of the promise spec, so unless you want to polyfill which I wouldn't do on lib level I wouldn't ue it 🤔

Comment (julianburr) [reply src/react-forms.js]: > Are we just ignoring submit errors now

what do you mean?

Comment (aaronnuu) [reply src/react-forms.js]: Are we not just swallowing submit errors now with that `catch` block? I would imagine we still want to handle them in in our apps, or I'm just not getting how the errors propagate in this case...

My thoughts are 
```
try {
  ...
  const result = await this.executeSubmit();
  ...
} finally {
  this.blockSubmission = false
}
```

This way errors in the `handleSubmit` function won't be swallowed and we still unblock submission.

Comment (julianburr) [reply src/react-forms.js]: sounds good to me, can you change it? :blush:

Comment (aaronnuu) [reply src/react-forms.js]: Yep will do

**Pull request merged**